lib/puppet/pops/lookup/interpolation.rb

```diff
@@ -90,14 +90,16 @@ def interpolate_method(method_key)
             lookup_invocation.report_found_in_overrides(root_key, ovr[root_key])
           else
             scope = lookup_invocation.scope
-            val = scope[root_key]
-            if val.nil? && !nil_in_scope?(scope, root_key)
-              defaults = lookup_invocation.default_values
-              if defaults.include?(root_key)
-                lookup_invocation.report_found_in_defaults(root_key, defaults[root_key])
-              else
-                nil
-              end
+            val = nil
+            if (default_val = lookup_invocation.default_values[root_key])
```

Comment:
I think we have to check if `default_values.include?(root_key)` because `default_val` may be `false`.